chainerx_cc/chainerx/native/native_device/comparison.cc

```diff
@@ -110,6 +111,51 @@ class NativeLogicalNotOp : public LogicalNotOp {
 
 CHAINERX_REGISTER_OP_NATIVE(LogicalNotOp, NativeLogicalNotOp);
 
+class NativeAllOp : public AllOp {
+public:
+    void Call(const Array& a, const Axes& axis, const Array& out) override {
+        CHAINERX_ASSERT(internal::IsValidReductionShape(a.shape(), axis, out.shape(), true));
+        a.device().CheckDevicesCompatible(a, out);
+        const Array& a_cast = a.dtype() == out.dtype() ? a : a.AsType(out.dtype());
+        auto do_all = [&a_cast, &axis, &out](auto in_pt) {
+            using In = typename decltype(in_pt)::type;
+            struct Impl {
+                bool Identity() { return true; }
+                bool MapIn(In in, int64_t /*index*/) { return static_cast<bool>(in); }
+                void Reduce(bool next, bool& accum) { accum = accum && next; }
+                bool MapOut(bool accum) { return static_cast<bool>(accum); }
```

Comment:
This cast seems not needed.
```suggestion
                bool MapOut(bool accum) { return accum; }
```

Reply:
Have removed the cast.